BSP/STM32F413/Src/BSP_CAN.c

```diff
@@ -125,10 +128,11 @@ uint8_t BSP_CAN_Write(uint32_t id, uint8_t data[8], uint8_t length) {
 uint8_t BSP_CAN_Read(uint32_t *id, uint8_t *data) {
     if(RxFlag){
 		for(int i = 0; i < 8; i++){
-			data[i] = RxMessage.Data[i];
+			data[i] = rxFifo[rxFifoTail][i];
+            (++rxFifoTail)%CAN_FIFO_SIZE; //increment FIFO tail
 		}
```

Comment:
Why increment the tail from inside the loop? If you have 8 messages in the buffer, you're going to read a little bit of all of them.

Reply:
Moved outside loop